Cannot compile rustc-ap-rustc_data_structures on nightly

```
error[E0061]: this function takes 1 argument but 0 arguments were supplied
  --> /Users/dmalyshau/.cargo/registry/src/github.com-1ecc6299db9ec823/rustc-ap-rustc_data_structures-637.0.0/box_region.rs:34:58
   |
34 |         let init = match Pin::new(&mut result.generator).resume() {
   |                                                          ^^^^^^- supplied 0 arguments
   |
help: expected the unit value `()`; create it with empty parentheses
   |
34 |         let init = match Pin::new(&mut result.generator).resume(()) {
   |                                                                 ^^

error[E0061]: this function takes 1 argument but 0 arguments were supplied
  --> /Users/dmalyshau/.cargo/registry/src/github.com-1ecc6299db9ec823/rustc-ap-rustc_data_structures-637.0.0/box_region.rs:48:76
   |
48 |         if let GeneratorState::Complete(_) = Pin::new(&mut self.generator).resume() {
   |                                                                            ^^^^^^- supplied 0 arguments
   |
help: expected the unit value `()`; create it with empty parentheses
   |
48 |         if let GeneratorState::Complete(_) = Pin::new(&mut self.generator).resume(()) {
   |                                                                                   ^^

error[E0061]: this function takes 1 argument but 0 arguments were supplied
  --> /Users/dmalyshau/.cargo/registry/src/github.com-1ecc6299db9ec823/rustc-ap-rustc_data_structures-637.0.0/box_region.rs:57:52
   |
57 |         let result = Pin::new(&mut self.generator).resume();
   |                                                    ^^^^^^- supplied 0 arguments
   |
help: expected the unit value `()`; create it with empty parentheses
   |
57 |         let result = Pin::new(&mut self.generator).resume(());
   |                                                           ^^

error: aborting due to 3 previous errors

For more information about this error, try `rustc --explain E0061`.
error: could not compile `rustc-ap-rustc_data_structures`.
```

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -27,6 +27,8 @@ json = "0.12.1"
 petgraph = "0.5.0"
 structopt = "0.3.7"
 
-syntax = { package = "rustc-ap-syntax", version = "637.0.0" }
-rustc_span = { package = "rustc-ap-rustc_span", version = "637.0.0" }
-rustc_parse = { package = "rustc-ap-rustc_parse", version = "637.0.0" }
+rustc_ast = { package = "rustc-ap-rustc_ast", version = "657" }
+rustc_ast_pretty = { package = "rustc-ap-rustc_ast_pretty", version = "657" }
+rustc_span = { package = "rustc-ap-rustc_span", version = "657" }
+rustc_parse = { package = "rustc-ap-rustc_parse", version = "657" }
+rustc_session = { package = "rustc-ap-rustc_session", version = "657" }

--- a/src/builder.rs
+++ b/src/builder.rs
@@ -1,7 +1,7 @@
 use std::{ffi::OsStr, fs, io, path, result::Result, string::ToString};
 
-use syntax::print::pprust;
-use syntax::{ast, visit};
+use rustc_ast_pretty::pprust;
+use rustc_ast::{ast, visit};
 
 use rustc_span::source_map;
 
@@ -148,7 +148,7 @@ impl<'a> visit::Visitor<'a> for Builder<'a> {
         }
     }
 
-    fn visit_mac(&mut self, mac: &'a ast::Mac) {
+    fn visit_mac(&mut self, mac: &'a ast::MacCall) {
         visit::walk_mac(self, mac);
     }
 }

--- a/src/main.rs
+++ b/src/main.rs
@@ -6,9 +6,11 @@ extern crate json;
 extern crate petgraph;
 extern crate structopt;
 
-extern crate syntax;
+extern crate rustc_ast;
+extern crate rustc_ast_pretty;
 extern crate rustc_span;
 extern crate rustc_parse;
+extern crate rustc_session;
 
 mod builder;
 mod dot_printer;
@@ -21,9 +23,10 @@ mod tree;
 use std::path;
 use std::process;
 
-use syntax::ast::{Crate, NodeId};
-use syntax::sess::ParseSess;
-use syntax::visit::Visitor;
+use rustc_ast::ast::{Crate, NodeId};
+use rustc_ast::visit::Visitor;
+
+use rustc_session::parse::ParseSess;
 
 use rustc_span::source_map::{self, edition::Edition};
 
@@ -126,7 +129,7 @@ fn run(args: &Arguments) -> Result<(), Error> {
         return run_2018(args, &manifest);
     }
 
-    syntax::with_globals(Edition::Edition2015, || {
+    rustc_ast::attr::with_globals(Edition::Edition2015, || {
         let parse_session = ParseSess::new(source_map::FilePathMapping::empty());
 
         let krate: Crate =

--- a/src/ng/analysis.rs
+++ b/src/ng/analysis.rs
@@ -10,17 +10,19 @@ use error::Error;
 use manifest::Target;
 use ng::graph::{Graph, GraphBuilder, Visibility, GLOB, SEP};
 
-use syntax::ast::{
-    Attribute, Crate, Item, ItemKind, Mac, Mod, NodeId, UseTree, UseTreeKind, VisibilityKind,
+use rustc_ast::ast::{
+    Attribute, Crate, Item, ItemKind, MacCall, Mod, NodeId, UseTree, UseTreeKind, VisibilityKind,
 };
-use syntax::print::pprust;
-use syntax::sess::ParseSess;
-// use syntax::source_map::{edition::Edition, FilePathMapping, SourceMap, Span, Symbol};
-use syntax::visit::{self, Visitor};
+// use rustc_ast::source_map::{edition::Edition, FilePathMapping, SourceMap, Span, Symbol};
+use rustc_ast::visit::{self, Visitor};
+
+use rustc_ast_pretty::pprust;
 
 use rustc_span::source_map::{edition::Edition, FilePathMapping, SourceMap, Span, Symbol};
 
 use rustc_parse;
+
+use rustc_session::parse::ParseSess;
 
 const SOURCE_DIR: &str = "./src/";
 const DIR_SEP: &str = "/";
@@ -74,7 +76,7 @@ pub fn build_graph<'a>(
     target: &Target,
     ignored_files: &'a [PathBuf],
 ) -> Result<Graph, Error> {
-    syntax::with_globals(edition, || {
+    rustc_ast::attr::with_globals(edition, || {
         let parse_session = ParseSess::new(FilePathMapping::empty());
         let crate_: Crate =
             match rustc_parse::parse_crate_from_file(&target.src_path(), &parse_session) {
@@ -185,7 +187,7 @@ impl<'a> Visitor<'a> for Builder<'a> {
         }
     }
 
-    fn visit_mac(&mut self, macro_: &'a Mac) {
+    fn visit_mac(&mut self, macro_: &'a MacCall) {
         visit::walk_mac(self, macro_);
     }
 

